provider/datagen/src/provider.rs

```diff
@@ -12,16 +12,16 @@ use std::fmt::Debug;
 use std::path::PathBuf;
 use std::sync::Arc;
 
-/// A [`DataProvider`] backed by raw CLDR and ICU data.
+/// An [`ExportableProvider`](icu_provider::datagen::ExportableProvider) backed by raw CLDR and ICU data.
 ///
 /// This provider covers all keys that are used by ICU4X. It is intended as the canonical
 /// provider for [`DatagenDriver::export`](crate::DatagenDriver::export).
 ///
-/// If a specific data source has not been set, `DataProvider::load` will
-/// error ([`is_missing_cldr_error`](crate::is_missing_cldr_error) /
-/// [`is_missing_icuexport_error`](crate::is_missing_icuexport_error)) /
-/// [`is_missing_segmenter_lstm_error`](crate::is_missing_segmenter_lstm_error))
-/// if the data is required for that key.
+/// If a required data source has not been set, `DataProvider::load` will
+/// fail with the appropriate error:
+/// * [`is_missing_cldr_error`](Self::is_missing_cldr_error)
+/// * [`is_missing_icuexport_error`](Self::is_missing_icuexport_error)
+/// * [`is_missing_segmenter_lstm_error`](Self::is_missing_segmenter_lstm_error)
```

Comment:
Why did you move these onto `DatagenProvider`? I think they were good as free functions in the error module

Reply:
Free functions are messy. These errors are a property of the DatagenProvider, not of any other part of datagen.